packages/core/test/interpreter.test.ts

```diff
@@ -1937,5 +1937,60 @@ Event: {\\"type\\":\\"SOME_EVENT\\"}"
         })
         .start();
     });
+
+    it('stopped spawned actors should be cleaned up in parent', (done) => {
+      const childMachine = Machine({
+        initial: 'idle',
+        states: {
+          idle: {}
+        }
+      });
+
+      const parentMachine = createMachine<any>({
+        id: 'form',
+        initial: 'present',
+        context: {},
+        entry: assign({
+          machineRef: () => spawn(childMachine, 'machineChild'),
+          promiseRef: () =>
+            spawn(
+              new Promise(() => {
+                // ...
+              }),
+              'promiseChild'
+            ),
+          observableRef: () => spawn(interval(1000), 'observableChild')
+        }),
+        states: {
+          present: {
+            on: { NEXT: 'gone' }
+          },
+          gone: {
+            type: 'final'
+          }
+        }
+      });
+
+      const service = interpret(parentMachine)
+        .onDone(() => {
+          expect(service.children.get('machineChild')).toBeUndefined();
+          expect(service.children.get('promiseChild')).toBeUndefined();
+          expect(service.children.get('observableChild')).toBeUndefined();
+          done();
```

Comment:
I'm not sure if this really should be allowed. It's basically bypassing the machinery of state machine, allowing for arbitrary, imperative, manipulation of its internal state. This also doesn't quite deliver proper cleanup:
```js
typeof service.state.context.machineRef === 'object'
typeof service.state.context.promiseRef === 'object'
typeof service.state.context. observableRef === 'object'
```

This is the common case - spawned actors are usually assigned to the context. Clearing context values would definitely be too magical (and not trivial - with the current APIs it would require a full crawl of the current context).